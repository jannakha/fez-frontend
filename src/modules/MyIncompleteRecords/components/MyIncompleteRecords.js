import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { StandardPage } from 'modules/SharedComponents/Toolbox/StandardPage';
import { StandardRighthandCard } from 'modules/SharedComponents/Toolbox/StandardRighthandCard';
import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';
import { InlineLoader } from 'modules/SharedComponents/Toolbox/Loaders';
import {
    PublicationsList,
    PublicationsListPaging,
    PublicationsListSorting,
    FacetsFilter
} from 'modules/SharedComponents/PublicationsList';
import locale from 'locale/components';
import { routes } from 'config';
import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';

export default class MyIncompleteRecords extends PureComponent {
    static propTypes = {
        incomplete: PropTypes.shape({
            publicationsList: PropTypes.array,
            publicationsListFacets: PropTypes.object,
            loadingPublicationsList: PropTypes.bool,
            publicationsListPagingData: PropTypes.object,
        }),
        exportPublicationsLoading: PropTypes.bool,

        initialFacets: PropTypes.object,
        accountLoading: PropTypes.bool,
        localePages: PropTypes.object,
        thisUrl: PropTypes.string,
        canUseExport: PropTypes.bool,

        location: PropTypes.object.isRequired,
        history: PropTypes.object.isRequired,
        actions: PropTypes.object
    };

    constructor(props) {
        super(props);

        this.initState = {
            page: 1,
            pageSize: 20,
            sortBy: locale.components.sorting.sortBy[0].value,
            sortDirection: locale.components.sorting.sortDirection[0],
            activeFacets: {
                filters: {},
                ranges: {},
                ...props.initialFacets
            }
        };

        this.state = {
            // check if user has publications, once true always true
            // facets filtering might return no results, but facets should still be visible
            hasPublications: !props.incomplete.loadingPublicationsList && props.incomplete.publicationsList.length > 0,
            ...(!!props.location.state ? props.location.state : this.initState)
        };
    }

    componentDidMount() {
        if (!this.props.accountLoading && !this.props.incomplete.publicationsList.length) {
            this.props.actions.searchAuthorPublications({ ...this.state }, 'incomplete');
        }
    }

    componentWillReceiveProps(newProps) {
        // handle browser back button - set state from location/dispatch action for this state
        if (this.props.location !== newProps.location
            && newProps.history.action === 'POP'
            && newProps.location.pathname === this.props.thisUrl) {
            this.setState({ ...(
                !!newProps.location.state
                    ? newProps.location.state
                    : this.initState
            )}, () => {
                // only will be called when user clicks back on my records page
                this.props.actions.searchAuthorPublications({ ...this.state }, 'incomplete');
            });
        }
        // set forever-true flag if user has publications
        if (!this.state.hasPublications && !newProps.incomplete.loadingPublicationsList
            && !!newProps.incomplete.publicationsList && newProps.incomplete.publicationsList.length > 0) {
            this.setState({ hasPublications: true });
        }
    }

    pageSizeChanged = (pageSize) => {
        this.setState(
            {
                pageSize: pageSize,
                page: 1
            }, this.pushPageHistory
        );
    }

    pageChanged = (page) => {
        this.setState(
            {
                page: page
            }, this.pushPageHistory
        );
    }

    sortByChanged = (sortBy, sortDirection) => {
        this.setState(
            {
                sortBy: sortBy,
                sortDirection: sortDirection
            }, this.pushPageHistory
        );
    }

    facetsChanged = (activeFacets) => {
        this.setState(
            {
                activeFacets: activeFacets,
                page: 1
            }, this.pushPageHistory
        );
    }

    pushPageHistory = () => {
        this.props.history.push({
            pathname: `${this.props.thisUrl}`,
            search: `?ts=${Date.now()}`,
            state: { ...this.state }
        });
        this.props.actions.searchAuthorPublications({ ...this.state }, 'incomplete');
    };

    completeRecord = (item) => {
        this.props.history.push(routes.pathConfig.records.incompleteFix(item.rek_pid));
    }

    render() {
        if (this.props.accountLoading) return null;
        const txt = this.props.localePages;
        const pagingData = this.props.incomplete.publicationsListPagingData;
        const isLoading = this.props.incomplete.loadingPublicationsList;
        const inProgress = [
            {
                label: 'In progress',
                disabled: true,
                primary: false
            }
        ];

        const actions = [
            {
                label: txt.completeRecordButton,
                handleAction: this.completeRecord,
                primary: true
            }
        ];

        return (
            <StandardPage title={txt.pageTitle}>
                <Grid container spacing={16}>
                    {
                        // first time loading my publications - account hasn't been loaded or
                        // any my publications haven't been loaded
                        !this.state.hasPublications &&
                        this.props.incomplete.loadingPublicationsList &&
                        <Grid item xs={12}>
                            <InlineLoader message={txt.loadingMessage} />
                        </Grid>
                    }
                    {
                        // no results to display
                        !this.props.incomplete.loadingPublicationsList &&
                        this.props.incomplete.publicationsList &&
                        this.props.incomplete.publicationsList.length === 0 &&
                        <Grid item xs={12}>
                            <StandardCard {...txt.noResultsFound}>
                                {txt.noResultsFound.text}
                            </StandardCard>
                        </Grid>
                    }
                    {
                        // results to display or loading if user is filtering/paging
                        this.state.hasPublications && (
                            this.props.incomplete.loadingPublicationsList ||
                            this.props.incomplete.publicationsList.length > 0
                        ) &&
                        <Grid item xs={12} md={9}>
                            <StandardCard noHeader>
                                {
                                    pagingData &&
                                    pagingData.to &&
                                    pagingData.from &&
                                    pagingData.total &&
                                    <span>
                                        {txt.recordCount
                                            .replace('[recordsTotal]', pagingData.total)
                                            .replace('[recordsFrom]', pagingData.from)
                                            .replace('[recordsTo]', pagingData.to)}
                                    </span>
                                }
                                <Grid container spacing={16}>
                                    <Grid item xs={12}>
                                        {txt.text}
                                    </Grid>
                                    <Grid item xs={12}>
                                        <PublicationsListSorting
                                            initPageLength={this.initState.pageSize}
                                            sortBy={this.state.sortBy}
                                            sortDirection={this.state.sortDirection}
                                            pageSize={this.state.pageSize}
                                            pagingData={pagingData}
                                            onSortByChanged={this.sortByChanged}
                                            onPageSizeChanged={this.pageSizeChanged}
                                            disabled={isLoading} />
                                    </Grid>
                                    <Grid item xs={12}>
                                        <PublicationsListPaging
                                            loading={isLoading}
                                            pagingData={pagingData}
                                            onPageChanged={this.pageChanged}
                                            disabled={isLoading} />
                                    </Grid>
                                    <Grid item xs={12}>
                                        {
                                            isLoading &&
                                            <div className="is-centered">
                                                <InlineLoader message={
                                                    this.props.incomplete.loadingPublicationsList &&
                                                    txt.loadingPagingMessage
                                                } />
                                            </div>
                                        }
                                        {
                                            !this.props.exportPublicationsLoading &&
                                            !this.props.incomplete.loadingPublicationsList &&
                                            this.props.incomplete.publicationsList &&
                                            this.props.incomplete.publicationsList.length > 0 &&
                                            <PublicationsList
                                                publicationsList={this.props.incomplete.publicationsList}
                                                subsetCustomActions={inProgress}
                                                customActions={actions}
                                            />
                                        }
                                    </Grid>
                                    <Grid item xs={12}>
                                        <PublicationsListPaging
                                            loading={isLoading}
                                            pagingData={pagingData}
                                            onPageChanged={this.pageChanged}
                                            disabled={isLoading} />
                                    </Grid>
                                </Grid>
                            </StandardCard>
                        </Grid>
                    }
                    {
                        // show available filters or selected filters (even if there are no results)
                        ((this.props.incomplete.publicationsListFacets && Object.keys(this.props.incomplete.publicationsListFacets).length > 0)
                            || (this.state.activeFacets && this.state.activeFacets.ranges && Object.keys(this.state.activeFacets.ranges).length > 0)
                            || (this.state.activeFacets && !!this.state.activeFacets.showOpenAccessOnly)) &&
                        <Hidden smDown>
                            <Grid item md={3}>
                                <StandardRighthandCard title={txt.facetsFilter.title} help={txt.facetsFilter.help}>
                                    <FacetsFilter
                                        facetsData={this.props.incomplete.publicationsListFacets}
                                        onFacetsChanged={this.facetsChanged}
                                        activeFacets={this.state.activeFacets}
                                        disabled={isLoading}
                                        excludeFacetsList={txt.facetsFilter.excludeFacetsList}
                                        renameFacetsList={txt.facetsFilter.renameFacetsList}
                                        lookupFacetsList={txt.facetsFilter.lookupFacetsList}
                                        showOpenAccessFilter />
                                </StandardRighthandCard>
                            </Grid>
                        </Hidden>
                    }
                </Grid>
            </StandardPage>
        );
    }
}
