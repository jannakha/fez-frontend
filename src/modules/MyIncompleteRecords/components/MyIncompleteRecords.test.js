import MyIncompleteRecords from './MyIncompleteRecords';
import { routes, general } from 'config';
import { locale } from 'locale';
import { mockRecordToFix } from 'mock/data/testing/records';

function setup(testProps, isShallow = true) {
    const props = {
        actions: {
            searchAuthorPublications: jest.fn(),
            setFixRecord: jest.fn(),
        },
        location: {
            pathname: routes.pathConfig.records.mine,
            state: null
        },
        history: {
            push: jest.fn(),
            go: jest.fn()
        },
        accountLoading: false,
        exportPublicationsLoading: false,
        localePages: locale.pages.myResearch,
        ...testProps,
        incomplete: {
            publicationsListPagingData: {},
            loadingPublicationsList: false,
            publicationsList: [],
            publicationsListFacets: {},
            ...testProps.incomplete
        },
    };
    return getElement(MyIncompleteRecords, props, isShallow);
}

describe('MyIncompleteRecords test', () => {
    it('renders loading screen while loading account data', () => {
        const wrapper = setup({ accountLoading: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders loading screen while loading publications ', () => {
        const wrapper = setup({ incomplete:
            {
                loadingPublicationsList: true
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders loading screen while loading publications while filtering', () => {
        const wrapper = setup({
            incomplete: {
                publicationsList: [1, 2, 2]
            }
        });
        wrapper.setProps({
            incomplete: {
                loadingPublicationsList: true
            }
        });
        wrapper.update();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders loading screen while export publications loading', () => {
        const wrapper = setup({
            incomplete: {
                publicationsList: [1, 2, 2],
            },
            exportPublicationsLoading: true
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders no results', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders list of publications no facets', () => {
        const wrapper = setup({
            incomplete: {
                publicationsList: [1, 2, 3], // myRecordsList.data,
                publicationsListPagingData: {
                    "total": 147,
                    "per_page": 20,
                    "current_page": 1,
                    "from": 1,
                    "to": 20
                }
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders list of publications with facets', () => {
        const wrapper = setup({
            incomplete: {
                publicationsList: [1, 2, 3], // myRecordsList.data,
                publicationsListPagingData: {
                    "total": 147,
                    "per_page": 20,
                    "current_page": 1,
                    "from": 1,
                    "to": 20
                },
                publicationsListFacets: {
                    "Display type": {
                        "doc_count_error_upper_bound": 0,
                        "sum_other_doc_count": 3,
                        "buckets": [
                            { "key": 179, "doc_count": 95 },
                            { "key": 130, "doc_count": 34 },
                            { "key": 177, "doc_count": 2 },
                            { "key": 183, "doc_count": 2 },
                            { "key": 174, "doc_count": 1 }
                        ]
                    },
                    "Keywords": {
                        "doc_count_error_upper_bound": 0,
                        "sum_other_doc_count": 641,
                        "buckets": [
                            { "key": "Brca1", "doc_count": 15 },
                            { "key": "Oncology", "doc_count": 15 },
                            { "key": "Breast cancer", "doc_count": 13 },
                            { "key": "Genetics & Heredity", "doc_count": 12 },
                            { "key": "Biochemistry & Molecular Biology", "doc_count": 10 }
                        ]
                    }
                }
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders active filters', () => {
        const wrapper = setup({
            location: {
                state: {
                    activeFacets: {
                        filters: {},
                        ranges: {
                            Year: {
                                from: 2000,
                                to: 2010
                            }
                        }
                    }
                }
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('state is updated by sub components', () => {
        const testAction = jest.fn();
        const wrapper = setup({
            actions: {
                searchAuthorPublications: testAction
            }
        });

        wrapper.instance().pageSizeChanged(100);
        expect(wrapper.state().pageSize).toEqual(100);
        expect(wrapper.state().page).toEqual(1);
        expect(testAction).toHaveBeenCalled();

        wrapper.instance().pageChanged(2);
        expect(wrapper.state().page).toEqual(2);
        expect(testAction).toHaveBeenCalled();

        wrapper.instance().sortByChanged('foo', 'bar');
        expect(wrapper.state().sortBy).toEqual('foo');
        expect(wrapper.state().sortDirection).toEqual('bar');
        expect(testAction).toHaveBeenCalled();

        wrapper.instance().facetsChanged({ 'foo': 'bar' });
        expect(wrapper.state().activeFacets).toEqual({ 'foo': 'bar' });
        expect(wrapper.state().page).toEqual(1);
        expect(testAction).toHaveBeenCalled();
    });

    it('sets forever true has publications on load', () => {
        const wrapper = setup({
            location: {
                state: {
                    page: 2,
                    hasPublications: true
                }
            }
        });
        expect(wrapper.state().hasPublications).toEqual(true);
        expect(wrapper.state().page).toEqual(2);
    });

    it('sets forever true has publications', () => {
        const wrapper = setup({ incomplete: { loadingPublicationsList: true }});
        expect(wrapper.state().hasPublications).toEqual(false);

        wrapper.instance().componentWillReceiveProps({
            incomplete: {
                loadingPublicationsList: false,
                publicationsList: [1, 2, 3],
            },
            history: {},
            location: {}
        });
        expect(wrapper.state().hasPublications).toEqual(true);
    });

    it('gets publications when user clicks back and state is set', () => {
        const testAction = jest.fn();
        const wrapper = setup({
            accountLoading: true,
            actions: {
                searchAuthorPublications: testAction
            },
            thisUrl: routes.pathConfig.records.mine
        });

        wrapper.instance().componentWillReceiveProps({
            history: { action: 'POP' },
            location: {
                pathname: routes.pathConfig.records.mine,
                state: {
                    page: 2,
                    hasPublications: true
                }
            }
        });
        expect(testAction).toHaveBeenCalled();
        expect(wrapper.state().hasPublications).toEqual(true);
        expect(wrapper.state().page).toEqual(2);

    });

    it('gets publications when user clicks back and state is not set', () => {
        const testAction = jest.fn();
        const wrapper = setup({
            accountLoading: true,
            actions: {
                searchAuthorPublications: testAction
            },
            thisUrl: routes.pathConfig.records.mine
        });
        wrapper.instance().componentWillReceiveProps({
            history: { action: 'POP' },
            location: {
                pathname: routes.pathConfig.records.mine,
                state: null
            },
            incomplete: {
                loadingPublicationsList: false,
                publicationsList: [],
            }
        });
        expect(testAction).toHaveBeenCalled();
        expect(wrapper.state().page).toEqual(1);
    });

    it('doesn\'t retrieve data from history if user navigates to next page', () => {
        const testAction = jest.fn();
        const wrapper = setup({
            accountLoading: true,
            actions: {
                searchAuthorPublications: testAction
            }
        });

        wrapper.instance().componentWillReceiveProps({
            history: { action: 'PUSH' },
            location: { pathname: routes.pathConfig.records.mine },
            incomplete: {
                loadingPublicationsList: false,
                publicationsList: [],
            }
        });
        expect(testAction).not.toHaveBeenCalled();
    });

    // it('sets publication to fix', () => {
    //     const push = jest.fn();
    //     const setFixRecord = jest.fn();
    //     const wrapper = setup({accountLoading: true, actions: {setFixRecord: setFixRecord},  history: {push: push}});
    //     wrapper.instance().completeRecord({rek_pid: 'UQ:111111'});
    //     expect(push).toHaveBeenCalledWith(routes.pathConfig.records.incompleteFix('UQ:111111'));
    //     expect(setFixRecord).toHaveBeenCalled();
    // });

    it('component has displayable facets', () => {
        const testAction = jest.fn();
        const wrapper = setup({ actions: { searchAuthorPublications: testAction } });

        wrapper.instance().facetsChanged({
            'filters': {
                'Display type': general.PUBLICATION_TYPE_CREATIVE_WORK
            }
        });

        expect(wrapper.state().activeFacets).toEqual({
            "filters": {
                "Display type": general.PUBLICATION_TYPE_CREATIVE_WORK
            }
        });
        expect(wrapper.state().page).toEqual(1);
        expect(testAction).toHaveBeenCalled();
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should complete record', () => {
        const testMethod = jest.fn();
        const goBack = jest.fn();

        const wrapper = setup({
            recordToFix: mockRecordToFix,
            history: {
                push: testMethod,
                goBack: goBack
            }
        });
        const item = { rek_pid: 'UQ:1001' };
        wrapper.instance().completeRecord(item);
        expect(testMethod).toHaveBeenCalledWith('/records/UQ:1001/incomplete');
    });
});
