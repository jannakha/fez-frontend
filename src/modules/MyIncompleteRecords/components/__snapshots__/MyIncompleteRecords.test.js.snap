// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyIncompleteRecords test component has displayable facets 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <StandardCard
        text={
          <div>
            We were unable to find any results. You may be able to 
            <a
              href="/records/possible"
            >
              claim works we think may be yours
            </a>
             or 
            <a
              href="/records/add/find"
            >
              add a missing publication
            </a>
          </div>
        }
        title="No works found"
      >
        <div>
          We were unable to find any results. You may be able to 
          <a
            href="/records/possible"
          >
            claim works we think may be yours
          </a>
           or 
          <a
            href="/records/add/find"
          >
            add a missing publication
          </a>
        </div>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders active filters 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <StandardCard
        text={
          <div>
            We were unable to find any results. You may be able to 
            <a
              href="/records/possible"
            >
              claim works we think may be yours
            </a>
             or 
            <a
              href="/records/add/find"
            >
              add a missing publication
            </a>
          </div>
        }
        title="No works found"
      >
        <div>
          We were unable to find any results. You may be able to 
          <a
            href="/records/possible"
          >
            claim works we think may be yours
          </a>
           or 
          <a
            href="/records/add/find"
          >
            add a missing publication
          </a>
        </div>
      </StandardCard>
    </WithStyles(Grid)>
    <Hidden
      implementation="js"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={false}
      smDown={true}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Grid)
        item={true}
        md={3}
      >
        <StandardRighthandCard
          title="Refine results"
        >
          <FacetsFilter
            activeFacets={
              Object {
                "filters": Object {},
                "ranges": Object {
                  "Year": Object {
                    "from": 2000,
                    "to": 2010,
                  },
                },
              }
            }
            disabled={false}
            excludeFacetsList={
              Array [
                "Scopus document type",
                "Genre",
                "Year published",
              ]
            }
            facetsData={Object {}}
            lookupFacetsList={
              Object {
                "Author": "Author (lookup)",
                "Collection": "Collection (lookup)",
                "Subject": "Subject (lookup)",
              }
            }
            onFacetsChanged={[Function]}
            renameFacetsList={
              Object {
                "Display type": "Work type",
                "Subtype": "Work subtype",
              }
            }
            showOpenAccessFilter={true}
          />
        </StandardRighthandCard>
      </WithStyles(Grid)>
    </Hidden>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders list of publications no facets 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <span>
          Displaying records 1 to 20 of 147 total records. 
        </span>
        <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/records/possible"
              >
                claiming a work
              </a>
               or 
              <a
                href="/records/add/find"
              >
                adding a missing work
              </a>
              .
            </span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListSorting)
              disabled={false}
              initPageLength={20}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
              sortBy="published_date"
              sortDirection="Desc"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <PublicationsList
              customActions={
                Array [
                  Object {
                    "handleAction": [Function],
                    "label": undefined,
                    "primary": true,
                  },
                ]
              }
              hideCountDiff={false}
              publicationsList={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              publicationsListSubset={Array []}
              showDefaultActions={false}
              showMetrics={false}
              showSourceCountIcon={false}
              showSources={false}
              showUnpublishedBufferFields={false}
              subsetCustomActions={
                Array [
                  Object {
                    "disabled": true,
                    "label": "In progress",
                    "primary": false,
                  },
                ]
              }
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders list of publications with facets 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <span>
          Displaying records 1 to 20 of 147 total records. 
        </span>
        <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/records/possible"
              >
                claiming a work
              </a>
               or 
              <a
                href="/records/add/find"
              >
                adding a missing work
              </a>
              .
            </span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListSorting)
              disabled={false}
              initPageLength={20}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
              sortBy="published_date"
              sortDirection="Desc"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <PublicationsList
              customActions={
                Array [
                  Object {
                    "handleAction": [Function],
                    "label": undefined,
                    "primary": true,
                  },
                ]
              }
              hideCountDiff={false}
              publicationsList={
                Array [
                  1,
                  2,
                  3,
                ]
              }
              publicationsListSubset={Array []}
              showDefaultActions={false}
              showMetrics={false}
              showSourceCountIcon={false}
              showSources={false}
              showUnpublishedBufferFields={false}
              subsetCustomActions={
                Array [
                  Object {
                    "disabled": true,
                    "label": "In progress",
                    "primary": false,
                  },
                ]
              }
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={
                Object {
                  "current_page": 1,
                  "from": 1,
                  "per_page": 20,
                  "to": 20,
                  "total": 147,
                }
              }
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </StandardCard>
    </WithStyles(Grid)>
    <Hidden
      implementation="js"
      lgDown={false}
      lgUp={false}
      mdDown={false}
      mdUp={false}
      smDown={true}
      smUp={false}
      xlDown={false}
      xlUp={false}
      xsDown={false}
      xsUp={false}
    >
      <WithStyles(Grid)
        item={true}
        md={3}
      >
        <StandardRighthandCard
          title="Refine results"
        >
          <FacetsFilter
            activeFacets={
              Object {
                "filters": Object {},
                "ranges": Object {},
              }
            }
            disabled={false}
            excludeFacetsList={
              Array [
                "Scopus document type",
                "Genre",
                "Year published",
              ]
            }
            facetsData={
              Object {
                "Display type": Object {
                  "buckets": Array [
                    Object {
                      "doc_count": 95,
                      "key": 179,
                    },
                    Object {
                      "doc_count": 34,
                      "key": 130,
                    },
                    Object {
                      "doc_count": 2,
                      "key": 177,
                    },
                    Object {
                      "doc_count": 2,
                      "key": 183,
                    },
                    Object {
                      "doc_count": 1,
                      "key": 174,
                    },
                  ],
                  "doc_count_error_upper_bound": 0,
                  "sum_other_doc_count": 3,
                },
                "Keywords": Object {
                  "buckets": Array [
                    Object {
                      "doc_count": 15,
                      "key": "Brca1",
                    },
                    Object {
                      "doc_count": 15,
                      "key": "Oncology",
                    },
                    Object {
                      "doc_count": 13,
                      "key": "Breast cancer",
                    },
                    Object {
                      "doc_count": 12,
                      "key": "Genetics & Heredity",
                    },
                    Object {
                      "doc_count": 10,
                      "key": "Biochemistry & Molecular Biology",
                    },
                  ],
                  "doc_count_error_upper_bound": 0,
                  "sum_other_doc_count": 641,
                },
              }
            }
            lookupFacetsList={
              Object {
                "Author": "Author (lookup)",
                "Collection": "Collection (lookup)",
                "Subject": "Subject (lookup)",
              }
            }
            onFacetsChanged={[Function]}
            renameFacetsList={
              Object {
                "Display type": "Work type",
                "Subtype": "Work subtype",
              }
            }
            showOpenAccessFilter={true}
          />
        </StandardRighthandCard>
      </WithStyles(Grid)>
    </Hidden>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders loading screen while export publications loading 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/records/possible"
              >
                claiming a work
              </a>
               or 
              <a
                href="/records/add/find"
              >
                adding a missing work
              </a>
              .
            </span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListSorting)
              disabled={false}
              initPageLength={20}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={Object {}}
              sortBy="published_date"
              sortDirection="Desc"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={Object {}}
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          />
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={false}
              loading={false}
              onPageChanged={[Function]}
              pagingData={Object {}}
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders loading screen while loading account data 1`] = `""`;

exports[`MyIncompleteRecords test renders loading screen while loading publications  1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(InlineLoader)
        message="Searching for your works"
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders loading screen while loading publications while filtering 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      md={9}
      xs={12}
    >
      <StandardCard
        noHeader={true}
      >
        <WithStyles(Grid)
          container={true}
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <span>
              Add to this list by 
              <a
                href="/records/possible"
              >
                claiming a work
              </a>
               or 
              <a
                href="/records/add/find"
              >
                adding a missing work
              </a>
              .
            </span>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListSorting)
              disabled={true}
              initPageLength={20}
              onPageSizeChanged={[Function]}
              onSortByChanged={[Function]}
              pageSize={20}
              pagingData={Object {}}
              sortBy="published_date"
              sortDirection="Desc"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={true}
              loading={true}
              onPageChanged={[Function]}
              pagingData={Object {}}
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <div
              className="is-centered"
            >
              <WithStyles(InlineLoader)
                message="Retrieving your works"
              />
            </div>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            xs={12}
          >
            <WithStyles(PublicationsListPaging)
              disabled={true}
              loading={true}
              onPageChanged={[Function]}
              pagingData={Object {}}
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;

exports[`MyIncompleteRecords test renders no results 1`] = `
<StandardPage
  title="My research"
>
  <WithStyles(Grid)
    container={true}
    spacing={16}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <StandardCard
        text={
          <div>
            We were unable to find any results. You may be able to 
            <a
              href="/records/possible"
            >
              claim works we think may be yours
            </a>
             or 
            <a
              href="/records/add/find"
            >
              add a missing publication
            </a>
          </div>
        }
        title="No works found"
      >
        <div>
          We were unable to find any results. You may be able to 
          <a
            href="/records/possible"
          >
            claim works we think may be yours
          </a>
           or 
          <a
            href="/records/add/find"
          >
            add a missing publication
          </a>
        </div>
      </StandardCard>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</StandardPage>
`;
